pkg/notify/notifier/wechat/wechat.go

```diff
@@ -21,34 +21,67 @@ import (
 )
 
 const (
-	DefaultApiURL           = "https://qyapi.weixin.qq.com/cgi-bin/"
-	DefaultSendTimeout      = time.Second * 3
-	ToUserBatchSize         = 1000
-	ToPartyBatchSize        = 100
-	ToTagBatchSize          = 100
-	AccessTokenInvalid      = 42001
-	DefaultTextTemplate     = `{{ template "nm.default.text" . }}`
-	DefaultMarkdownTemplate = `{{ template "nm.default.markdown" . }}`
-	MessageMaxSize          = 2048
-	DefaultExpires          = time.Hour * 2
+	DefaultApiURL                 = "https://qyapi.weixin.qq.com/cgi-bin/"
+	DefaultSendTimeout            = time.Second * 3
+	ToUserBatchSize               = 1000
+	ToPartyBatchSize              = 100
+	ToTagBatchSize                = 100
+	AccessTokenInvalid            = 42001
+	DefaultTextTemplate           = `{{ template "nm.default.text" . }}`
+	DefaultMarkdownTemplate       = `{{ template "nm.default.markdown" . }}`
+	MessageMaxSize                = 2048
+	DefaultExpires                = time.Hour * 2
+	ChatbotMessageMaxTextSize     = 2048
+	ChatbotMessageMaxMarkdownSize = 4096
 )
 
 type Notifier struct {
-	notifierCtl    *controller.Controller
-	receiver       *wechat.Receiver
-	accessToken    string
-	timeout        time.Duration
-	logger         log.Logger
-	tmpl           *template.Template
-	ats            *notifier.AccessTokenService
-	messageMaxSize int
-	tokenExpires   time.Duration
+	notifierCtl        *controller.Controller
+	receiver           *wechat.Receiver
+	accessToken        string
+	timeout            time.Duration
+	logger             log.Logger
+	tmpl               *template.Template
+	ats                *notifier.AccessTokenService
+	messageMaxSize     int
+	tokenExpires       time.Duration
+	chatbotThreshold   int
+	chatbotUnit        time.Duration
+	chatbotMaxWaitTime time.Duration
 }
 
 type weChatMessageContent struct {
 	Content string `json:"content"`
 }
 
+type text struct {
+	Content             string   `json:"content,omitempty"`
+	MentionedList       []string `json:"mentioned_list,omitempty"`
+	MentionedMobileList []string `json:"mentioned_mobile_list,omitempty"`
+}
+
+type markdown struct {
+	Content string `json:"content,omitempty"`
+}
+
+type wechatBotText struct {
```

Comment:
I think we can expand the [weChatMessage](https://github.com/kubesphere/notification-manager/blob/master/pkg/notify/notifier/wechat/wechat.go#L52) to replace `wechatBotText` and `wechatBotMarkdown`.

Reply:
`wechatBotText` and `wechatBotMarkdown` are used to construct the json structure that the bot needs. Even if `weChatMessage` is extended, the json structure still needs to be constructed.

Reply:
The purpose of reusing existing structures is to make the code more concise.